File: day_1/day_1.py
# Problema 1: Achar duas entradas que somam 2020 e multiplicá-las
import pandas as pd
data = pd.read_csv("day_1/data.txt", sep = "\t")

# loop para achar as entradas
import random

soma = 1
while soma != 2020:
    numbers = data['expense'].sample(2)
    soma = numbers.sum()

File: day_1/day_1.R
# Problema 1: Achar duas entradas que somam 2020 e multiplicá-las
data = read.delim("day_1/data.txt")

# loop para achar as entradas
numbers = 1
while (sum(numbers) != 2020) {
    numbers = sample(data$expense, 2)
    sum(numbers)
}

# multiplicando
prod(numbers)

#################################################################

# Problema 2: Achar três entradas que somam 2020 e multiplicá-las

# loop para achar as entradas
numbers <- 1
while (sum(numbers) != 2020) {
    numbers <- sample(data$expense, 3)
    sum(numbers)
}

# multiplicando
prod(numbers)